plan.go

```diff
@@ -2837,17 +2839,20 @@ func (r *emptyFieldsPlan) do(ctx *execCtx, f func(id interface{}, data []interfa
 
 type wrapFilterPlan struct {
 	*filterDefaultPlan
+	once  sync.Once
```

Comment:
> can you expand more about what you think execCtx is good here.

Consider

```go
// error handling omitted
s, _ := db.Prepare("select * from t where exists (select * from t where x == ?)")
go s.Exec(1)
go s.Exec(4)
```

I think there will be only one plan instance, but two execCtx instances in the concurrently executing goroutines.

> somehow we will need to cache the value of matched. I'm wondering where that value could stay.

Correct and I think execCtx is the right place to extend with a bool field for that value. Or perhaps it should be `map[*selectStmt]bool` because the EXIST clause can appear more than once in a statement.